pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -18,16 +18,24 @@ def _VSCODE_convertNumpyArrayToDataFrame(ndarray):
     try:
         x_len = temp.shape[0]
         # Figure out if we're dealing with ragged data
-        # Handle ragged arrays by making a container where the number of
-        # columns is the max length of all rows. Missing elements are
-        # represented as empty strings.
         row_lengths = [
             len(temp[i])
             if isinstance(temp[i], (list, tuple, _VSCODE_np.ndarray))
             else 1
             for i in range(x_len)
         ]
-        flattened = _VSCODE_np.full((x_len, max(row_lengths)), "", dtype="object")
+        max_y_len = max(row_lengths)
+        is_ragged_or_ndimensional = hasattr(temp, "ndim") and (
+            (temp.ndim == 1 and max_y_len > 1) or temp.ndim > 2
+        )
+
+        if not is_ragged_or_ndimensional:
+            return _VSCODE_pd.DataFrame(temp)
```

Comment:
This is the key change--do a direct conversion instead of filling out a new ndarray of type object if the data is not ragged or ndimensional.